statsmodels/base/model.py

```diff
@@ -797,22 +797,24 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
         """
         import pandas as pd
 
-        exog_index = exog.index if _is_using_pandas(exog, None) else None
+        is_pandas = _is_using_pandas(exog, None)
 
-        if transform and hasattr(self.model, 'formula') and exog is not None:
+        exog_index = exog.index if is_pandas else None
+
+        if transform and hasattr(self.model, 'formula') and (exog is not None):
             from patsy import dmatrix
-            exog = pd.DataFrame(exog)  # user may pass series, if one predictor
```

Comment:
dropping this conversion, removes cases that can be converted to DataFrame, but that patsy cannot handle, e.g.
DataFrame.to_dict builds a nested dict including the index.